AWS Parameter Store SEPS Wallet Provider : If SEPS wallet is corrupted, the exception is not proper

**AWS Parameter Store SEPS Wallet Provider** : If SEPS wallet is corrupted, the exception is not proper.
It throws : java.lang.IllegalArgumentException: Last unit does not have enough valid bits
Without Specifying anything about wallet.

```

         java.lang.IllegalArgumentException: Last unit does not have enough valid bits
	at java.base/java.util.Base64$Decoder.decode0(Base64.java:766)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:538)
	at java.base/java.util.Base64$Decoder.decode(Base64.java:561)
	at oracle.jdbc.provider.aws.resource.ParameterStoreSEPSProvider.getWalletCredentials(ParameterStoreSEPSProvider.java:116)
	at oracle.jdbc.provider.aws.resource.ParameterStoreSEPSProvider.getPassword(ParameterStoreSEPSProvider.java:105)
	at oracle.provider.aws.resource.ParameterStoreSepsProvider1Test.testPkcs12PasswordWithoutIndex(ParameterStoreSepsProvider1Test.java:127)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at 
	...

```

The same test with **Secret Manager** throws : java.lang.IllegalStateException: Failed to open wallet
```
        
        java.lang.IllegalStateException: Failed to open wallet
	at oracle.jdbc.provider.util.WalletUtils.getCredentials(WalletUtils.java:191)
	at oracle.jdbc.provider.aws.resource.SecretsManagerSEPSProvider.getWalletCredentials(SecretsManagerSEPSProvider.java:111)
	at oracle.jdbc.provider.aws.resource.SecretsManagerSEPSProvider.getPassword(SecretsManagerSEPSProvider.java:93)
	at oracle.provider.aws.resource.SecretsManagerSepsProviderTest.testPkcs12PasswordWithoutIndex(SecretsManagerSepsProviderTest.java:137)
	
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	ExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	
Caused by: java.io.IOException: End of stream reached before all 6037 expected bytes could be read. Only received 1718 bytes before the end of the stream was reached
	at oracle.security.pki.OracleWallet.setWalletArray(Unknown Source)
	at oracle.security.pki.OracleWallet.setWalletArray(Unknown Source)
	at oracle.jdbc.provider.util.WalletUtils.getCredentials(WalletUtils.java:188)
	... 73 more
```



**To reproduce**

1.  Create a entry in Parameter Store with incomplete SEPS wallet
2. We can use existing test https://github.com/oracle/ojdbc-extensions/blob/main/ojdbc-provider-aws/src/test/java/oracle/provider/aws/resource/ParameterStoreSepsProviderTest.java to reproduce the issue

--- a/ojdbc-provider-common/src/main/java/oracle/jdbc/provider/util/WalletUtils.java
+++ b/ojdbc-provider-common/src/main/java/oracle/jdbc/provider/util/WalletUtils.java
@@ -194,7 +194,11 @@ public final class WalletUtils {
       wallet.setWalletArray(walletBytes, walletPassword);
     }
     catch (IOException ioException) {
-      throw new IllegalStateException("Failed to open wallet", ioException);
+      String message = "Failed to open wallet. The wallet content may be corrupted or incomplete.";
+      if (walletPassword != null) {
+        message += " If a walletPassword is required and was provided, it may be incorrect.";
+      }
+      throw new IllegalStateException(message, ioException);
     }
 
     try {

